### PR TITLE
chore(upgrade): Upgrading dependencies and code to ng8

### PR DESCRIPTION
**Pull request by bmarti44**
This related to issue https://github.com/gund/ng-dynamic-component/issues/257

Let me know any other changes you'd like made, and I'll get it updated!

I have also opened a PR for https://github.com/wishtack/wishtack-steroids/issues/183 which depends on `ng-dynamic-component`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,21 +50,19 @@
...
+    "@angular/common": "~8.0.3",
+    "@angular/core": "~8.0.3",
     "rxjs": "^6.0.0",
     "tslib": "^1.9.0"
   },
   "devDependencies": {
-    "@angular/animations": "^7.1.3",
-    "@angular/common": "^7.1.3",
-    "@angular/compiler": "^7.1.3",
-    "@angular/compiler-cli": "^7.1.3",
```
Comment (gund) [inline package.json]: Please add `@angular/compiler-cli` back to devDependencies
```suggestion
    "@angular/compiler-cli": "^7.1.3",
```

File: package.json [hunk 2]
```diff
@@ -50,21 +50,19 @@
     "node": ">=6.0.0"
   },
   "peerDependencies": {
-    "@angular/common": "^7.0.0",
-    "@angular/core": "^7.0.0",
+    "@angular/common": "~8.0.3",
+    "@angular/core": "~8.0.3",
```
Comment (gund) [inline package.json]: Please set versions to `^8.0.0` for all `peerDependencies`
```suggestion
    "@angular/core": "^8.0.0",
```

File: package.json [hunk 3]
```diff
@@ -50,21 +50,19 @@
     "node": ">=6.0.0"
   },
   "peerDependencies": {
-    "@angular/common": "^7.0.0",
-    "@angular/core": "^7.0.0",
+    "@angular/common": "~8.0.3",
```
Comment (gund) [inline package.json]:
```suggestion
    "@angular/common": "^8.0.0",
```

File: package.json [hunk 4]
```diff
@@ -50,21 +50,19 @@
...
-    "@angular/common": "^7.1.3",
-    "@angular/compiler": "^7.1.3",
-    "@angular/compiler-cli": "^7.1.3",
-    "@angular/core": "^7.1.3",
-    "@angular/http": "^7.1.3",
-    "@angular/platform-browser": "^7.1.3",
-    "@angular/platform-browser-dynamic": "^7.1.3",
-    "@angular/platform-server": "^7.1.3",
+    "@angular/animations": "^8.0.3",
+    "@angular/common": "~8.0.3",
```
Comment (gund) [inline package.json]:
```suggestion
    "@angular/common": "^8.0.3",
```

File: package.json [hunk 5]
```diff
@@ -50,21 +50,19 @@
...
+    "@angular/compiler": "~8.0.3",
```
Comment (gund) [inline package.json]:
```suggestion
    "@angular/compiler": "^8.0.3",
```

File: package.json [hunk 6]
```diff
@@ -50,21 +50,19 @@
...
+    "@angular/core": "~8.0.3",
```
Comment (gund) [inline package.json]:
```suggestion
    "@angular/core": "^8.0.3",
```

File: package.json [hunk 7]
```diff
@@ -50,21 +50,19 @@
...
+    "@angular/platform-browser": "~8.0.3",
```
Comment (gund) [inline package.json]:
```suggestion
    "@angular/platform-browser": "^8.0.3",
```

File: package.json [hunk 8]
```diff
@@ -50,21 +50,19 @@
...
+    "@angular/platform-browser-dynamic": "~8.0.3",
```
Comment (gund) [inline package.json]:
```suggestion
    "@angular/platform-browser-dynamic": "^8.0.3",
```

File: package.json [hunk 9]
```diff
@@ -50,21 +50,19 @@
...
+    "@angular/platform-server": "~8.0.3",
```
Comment (gund) [inline package.json]:
```suggestion
    "@angular/platform-server": "^8.0.3",
```

File: package.json [hunk 10]
```diff
@@ -50,21 +50,20 @@
...
-    "@angular/compiler-cli": "^7.1.3",
-    "@angular/core": "^7.1.3",
-    "@angular/http": "^7.1.3",
-    "@angular/platform-browser": "^7.1.3",
-    "@angular/platform-browser-dynamic": "^7.1.3",
-    "@angular/platform-server": "^7.1.3",
+    "@angular/animations": "^8.0.3",
+    "@angular/common": "^8.0.3",
+    "@angular/compiler": "^8.0.3",
+    "@angular/compiler-cli": "^8.0.3",
```
Comment (bmarti44) [inline package.json]: @gund added back with `^8.0.3`

**Comment by bmarti44** [comment]
@gund thanks for the feedback! let me know if i've missed anything, or if there are any additional changes you'd like me to make.

**Comment by bmarti44** [comment]
@gund I upgraded a few more things while I was in there... Started to get a few failing tests, and had to tweak `@ViewChild` and some of the typings for `jest.fn<MockDirective, any>` with an updated jest version. I know these are a lot of changes, but hopefully it's working better... Let me know if you want me to roll it back to just typescript and that's totally fine.

File: package.json [hunk 11]
```diff
@@ -129,12 +128,24 @@
...
+      "pre-push": "npm-run-all -p lint test",
+      "precommit": "lint-staged"
+    }
+  },
   "jest": {
     "preset": "jest-preset-angular",
     "roots": [
       "<rootDir>/src"
     ],
-    "setupTestFrameworkScriptFile": "<rootDir>/src/setupJest.ts",
```
Comment (bmarti44) [inline package.json]: This was renamed to `setupFilesAfterEnv` with newer `jest`

File: package.json [hunk 12]
```diff
@@ -129,12 +128,24 @@
       "git add"
     ]
   },
+  "husky": {
```
Comment (bmarti44) [inline package.json]: `husky` was complaining that these hooks were in run scripts, so moved to this husky config block

File: package.json [hunk 13]
```diff
@@ -8,9 +8,6 @@
   "es2015": "dist/bundles/ng-dynamic-component.es2015.js",
   "typings": "dist/ng-dynamic-component.d.ts",
   "scripts": {
-    "commitmsg": "commitlint -E HUSKY_GIT_PARAMS",
```
Comment (bmarti44) [inline package.json]: moved all of this to the `husky` config block

File: src/dynamic/component-outlet-injector.directive.spec.ts [hunk 14]
```diff
@@ -10,10 +10,12 @@ import {
...
 
 @Component({
-  template: `<ng-container *ngComponentOutlet="comp"></ng-container>`,
+  template: `
+    <ng-container *ngComponentOutlet="comp"></ng-container>
+  `,
 })
 class TestComponent extends TestComponentBase {
-  @ViewChild(ComponentOutletInjectorDirective)
+  @ViewChild(ComponentOutletInjectorDirective, { static: false })
```
Comment (bmarti44) [inline src/dynamic/component-outlet-injector.directive.spec.ts]: This extra parameter is added in ng8. False checks after `ngAfterViewInit` and true checks on `ngOnInit`, the default will be set to false (as that's when, most of the time, you'll wanna check).

File: src/dynamic/dynamic-attributes.directive.spec.ts [hunk 15]
```diff
@@ -30,7 +30,12 @@ describe('DynamicAttributesDirective', () => {
     let fixture: ComponentFixture<TestComponent>;
 
     @Component({
-      template: `<ng-container [ngComponentOutlet]="comp" [ndcDynamicAttributes]="attrs"></ng-container>`,
+      template: `
```
Comment (bmarti44) [inline src/dynamic/dynamic-attributes.directive.spec.ts]: Formatting/linting only

File: src/dynamic/dynamic-attributes.directive.spec.ts [hunk 16]
```diff
@@ -216,7 +221,11 @@ describe('DynamicAttributesDirective', () => {
     let fixture: ComponentFixture<TestComponent>;
 
     @Component({
-      template: `<ng-container *ngComponentOutlet="comp; ndcDynamicAttributes: attrs"></ng-container>`,
+      template: `
```
Comment (bmarti44) [inline src/dynamic/dynamic-attributes.directive.spec.ts]: Formatting/linting only

File: src/dynamic/dynamic-directives.directive.spec.ts [hunk 17]
```diff
@@ -358,7 +358,7 @@ describe('Directive: DynamicDirectives', () => {
 
   describe('with `ngComponentOutlet`', () => {
     let fixture: ComponentFixture<TestComponent>;
-    let created = jest.fn<MockDirective>();
+    let created = jest.fn<MockDirective, any>();
```
Comment (bmarti44) [inline src/dynamic/dynamic-directives.directive.spec.ts]: newer version of jest requires either 0 or 2, so just added `any`

**Comment by bmarti44** [comment]
@gund Seeing some jasmine errors in the build, also the new version of codelyzer renamed some of the linting checks... will fix it up

**Comment by bmarti44** [comment]
@gund build is looking happier...

File: src/setupJest.ts [hunk 18]
```diff
@@ -1,4 +1,5 @@
 import 'jest-preset-angular';
+import 'jest-extended';
```
Comment (gund) [inline src/setupJest.ts]: Why do you need `jest-extended` exactly?

Comment (bmarti44) [reply src/setupJest.ts]: when before i started messing with the typings, i thought i needed this for some missing functions off of jest... now testing locally, everything seems to work without it, so will remove

**Comment by bmarti44** [comment]
removed `jest-extended`, correct typing is included so it's all working nicely

File: tslint.json [hunk 19]
```diff
@@ -89,9 +88,9 @@
...
 
-    "use-input-property-decorator": true,
-    "use-output-property-decorator": true,
-    "use-host-property-decorator": true,
+    "no-inputs-metadata-property": true,
+    "no-outputs-metadata-property": true,
+    "no-host-metadata-property": true,
     "no-input-rename": true,
     "no-output-rename": true,
     "use-life-cycle-interface": true,
```
Comment (gund) [inline tslint.json]: `use-life-cycle-interface` is renamed to `use-lifecycle-interface` ([CI log warn: `Could not find implementations for the following rules specified in the configuration: use-life-cycle-interface`](https://travis-ci.org/gund/ng-dynamic-component/jobs/552542056))

```suggestion
    "use-lifecycle-interface": true,
```

**Comment by bmarti44** [comment]
@gund ran the tests and linting and ci run scripts on my local, looking good!

**Comment by bmarti44** [comment]
@gund GREAT THANK YOU SO MUCH! i have some updates to make now on https://github.com/wishtack/wishtack-steroids/issues/183 will do that now.

**Pull request merged**